### PR TITLE
Fix immutability in Specification results

### LINKED ISSUE #407: Read specification changed when each_batch used to fetch events
Here https://github.com/RailsEventStore/rails_event_store/blob/master/ruby_event_store/lib/ruby_event_store/specification.rb#L118

Read in batched is making changes in Specification definition

### PR DESCRIPTION
**Pull request by mpraglowski**
Closes #407

### CODE REVIEW

File: ruby_event_store/lib/ruby_event_store/specification.rb [hunk 1]
```diff
@@ -121,7 +121,7 @@ def limit(count)
     def each_batch
       return to_enum(:each_batch) unless block_given?
 
-      repository.read(result.tap { |r| r.read_as = :batch }).each do |batch|
+      repository.read(in_batches(result.batch_size).result).each do |batch|
```
Comment (pawelpacana) [inline ruby_event_store/lib/ruby_event_store/specification.rb]: 👍

File: ruby_event_store/spec/projection_spec.rb [hunk 2]
```diff
@@ -196,16 +196,18 @@ module RubyEventStore
     end
 
     specify do
-      specification = Specification.new(InMemoryRepository.new, Mappers::Default.new)
-      expect(Specification).to receive(:new).at_least(:once).and_return(specification)
-      expect(specification).to receive(:in_batches).with(2).and_return(specification)
+      repository = event_store.send(:repository)
+      mapper     = event_store.send(:mapper)
+      expected_specification = Specification.new(repository, mapper).in_batches(2).result
+      expect(repository).to receive(:read).with(expected_specification).and_return([])
```
Comment (pawelpacana) [inline ruby_event_store/spec/projection_spec.rb]: :+1:

Comment (pawelpacana) [reply ruby_event_store/spec/projection_spec.rb]: although it is more `expected_result`

Comment (mpraglowski) [reply ruby_event_store/spec/projection_spec.rb]: That might be changed soon ;)

**Pull request merged**